### DISCUSSION TITLE
Dbeaver tutorial

### DISCUSSION THREAD

**Discussion by casadmack74** [dbeaver/dbeaver#11068] [General]
Hi,

My objective is to create a database and populate it with data. Currently I have managed to connect to a database. I am using MariaDB as the database and Dbeaver as the GUI editor. The steps that I have in mind is

Create an ERD diagram for the empty database.

Create actual the tables and their relationships

Populate it with data.

Would anyone be able to help me in this regards.

Thanks & Best Regards.

**Comment by WeaseleyRon** [comment]
Hello,

I'm trying to learn how to make graphical stuff with java, and I was curious as to whether it would be better to stop using the netbeans design tab to make my applications, or continue using them.

Would it even be worth the trouble of learning how to hard code it? Would anyone ever require me to hard code my application instead of using a tool as helpful as this?
 https://19216801.onl/

**Reply by p91paul** [reply]
This is a very different topic from the post you are replying to!
Anyway, netbeans provides tools to design swing applications, which is the default java toolkit.
However in a work environment:
* java GUIs are extremely rare and getting rarer. Most enterprise applications are webapps now, and some desktop applications too (Elecron)
* When a java gui is needed, you might end up in a shop using a different gui framework, and I believe netbeans only handles swing.

So while learning, hardcoding is useful because a different framework might have some resemblance to swing in its low level api (positioning and sizing widgets can only be done in so many ways), so you can reuse the skills you learn. Once you are familiar with the API, I believe there's no reason for you to avoid the netbeans facilities.

**Comment by p91paul** [comment]
There is some documentation on specific topics on the Wiki https://github.com/dbeaver/dbeaver/wiki
